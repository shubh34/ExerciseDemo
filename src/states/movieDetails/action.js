import { RSAA } from 'redux-api-middleware';
import endPoints from '../../configs/endPoints';


export const GET_MOVIE_DETAILS_REQUEST = 'GET_MOVIE_DETAILS_REQUEST';
export const GET_MOVIE_DETAILS_SUCCESS = 'GET_MOVIE_DETAILS_SUCCESS';
export const GET_MOVIE_DETAILS_FAILURE = 'GET_MOVIE_DETAILS_FAILURE';

export const RESET_MOVIE_DETAILS = 'RESET_MOVIE_DETAILS';

export const getMovieDetails = (movieName) => {
	const endPoint = endPoints.getMovieDetails;
	const url = endPoint.resolve(movieName);
	return {
		[RSAA]: {
			endpoint: url,
			method: endPoint.method,
			types: [GET_MOVIE_DETAILS_REQUEST, GET_MOVIE_DETAILS_SUCCESS, GET_MOVIE_DETAILS_FAILURE],
		},
	};
};

export const resetMovieDetails = () => ({
	type: RESET_MOVIE_DETAILS,
});
