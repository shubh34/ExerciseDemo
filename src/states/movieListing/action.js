import { RSAA } from 'redux-api-middleware';
import endPoints from '../../configs/endPoints';

export const GET_MOVIE_LIST_BY_CATEGORY_REQUEST = 'GET_MOVIE_LIST_BY_CATEGORY_REQUEST';
export const GET_MOVIE_LIST_BY_CATEGORY_SUCCESS = 'GET_MOVIE_LIST_BY_CATEGORY_SUCCESS';
export const GET_MOVIE_LIST_BY_CATEGORY_FAILURE = 'GET_MOVIE_LIST_BY_CATEGORY_FAILURE';

export const getMovieListByCategory = (categoryName) => {
	const endPoint = endPoints.getMovieListByCategory;
	const url = endPoint.resolve(categoryName);
	const success = { type: GET_MOVIE_LIST_BY_CATEGORY_SUCCESS, meta: { id: categoryName } };
	const failure = { type: GET_MOVIE_LIST_BY_CATEGORY_FAILURE, meta: { id: categoryName } };
	return {
		[RSAA]: {
			endpoint: url,
			method: endPoint.method,
			types: [GET_MOVIE_LIST_BY_CATEGORY_REQUEST, success, failure],
		},
	};
};
