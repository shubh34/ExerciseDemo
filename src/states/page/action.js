
export const GO_TO_PAGE = 'GO_TO_PAGE';


export const goToPage = name => ({
	type: GO_TO_PAGE,
	name,
});
