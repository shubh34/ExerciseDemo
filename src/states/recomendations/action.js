import { RSAA } from 'redux-api-middleware';
import endPoints from '../../configs/endPoints';

export const GET_MOVIE_RECOMENDATION_REQUEST = 'GET_MOVIE_RECOMENDATION_REQUEST';
export const GET_MOVIE_RECOMENDATION_SUCCESS = 'GET_MOVIE_RECOMENDATION_SUCCESS';
export const GET_MOVIE_RECOMENDATION_FAILURE = 'GET_MOVIE_RECOMENDATION_FAILURE';

export const fetchRecomendationForMovie = id => {
	const endPoint = endPoints.getRecomendationForMovie;
	const url = endPoint.resolve(id);
	return {
		[RSAA]: {
			endpoint: url,
			method: endPoint.method,
			headers: {
				Accept: 'application/json',
				'content-type': 'application/json'
			},
			types: [GET_MOVIE_RECOMENDATION_REQUEST, GET_MOVIE_RECOMENDATION_SUCCESS, GET_MOVIE_RECOMENDATION_FAILURE]
		}
	};
};
