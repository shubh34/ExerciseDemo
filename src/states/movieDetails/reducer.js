import { GET_MOVIE_DETAILS_SUCCESS } from './action';


const reducer = (state = {}, action) => {
	switch (action.type) {
	case GET_MOVIE_DETAILS_SUCCESS:
		return {
			...state,
			details: action.payload,
		};
	default:
		return state;
	}
};
export default reducer;
