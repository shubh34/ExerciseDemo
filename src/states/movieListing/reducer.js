import { GET_MOVIE_LIST_BY_CATEGORY_SUCCESS } from './action';


const reducer = (state = [], action) => {
	switch (action.type) {
	case GET_MOVIE_LIST_BY_CATEGORY_SUCCESS:
		console.log(action.payload);
	default:
		console.log('state', state);
		return state;
	}
};
export default reducer;
