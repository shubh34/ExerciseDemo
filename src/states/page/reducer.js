import { GO_TO_PAGE } from './action';

export const initailPage = {
	name: 'Home',
	backPage: 'Home',
};

const reducer = (state = initailPage, action) => {
	switch (action.type) {
	case GO_TO_PAGE:
		return {
			...state,
			name: action.name,
		};
	default:
		return state;
	}
};
export default reducer;
