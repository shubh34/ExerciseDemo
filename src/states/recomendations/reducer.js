import { GET_MOVIE_RECOMENDATION_REQUEST, GET_MOVIE_RECOMENDATION_SUCCESS } from './action';

const defaultState = {};
const reducer = (state = defaultState, action) => {
	switch (action.type) {
	case GET_MOVIE_RECOMENDATION_SUCCESS:
		return {
			...state,
			...action.payload,
		};
	case GET_MOVIE_RECOMENDATION_REQUEST:
		return defaultState;
	default:
		return state;
	}
};
export default reducer;
