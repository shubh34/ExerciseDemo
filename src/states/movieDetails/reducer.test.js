import reducer from './reducer';
import { GET_MOVIE_DETAILS_SUCCESS, GET_MOVIE_DETAILS_FAILURE, RESET_MOVIE_DETAILS } from './action';

const state = {
	details: {
		movieName: 'test'
	}
};
describe('Reducer movie details', () => {
	it('should update store on GET_MOVIE_DETAILS_SUCCESS', () => {
		const action = {
			type: GET_MOVIE_DETAILS_SUCCESS,
			payload: { movieName: 'matrix' }
		};
		const updateState = reducer(state, action);
		expect(updateState).toMatchSnapshot();
	});
	it('should update store on GET_MOVIE_DETAILS_FAILURE', () => {
		const action = {
			type: GET_MOVIE_DETAILS_FAILURE,
			payload: { error: 'something went wrong' }
		};
		const updateState = reducer(state, action);
		expect(updateState).toMatchSnapshot();
	});
	it('should reset store on RESET_MOVIE_DETAILS', () => {
		const action = {
			type: RESET_MOVIE_DETAILS
		};
		const updateState = reducer(state, action);
		expect(updateState).toMatchSnapshot();
	});
});
