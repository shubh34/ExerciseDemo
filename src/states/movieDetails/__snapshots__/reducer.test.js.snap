// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reducer movie details should update store on GET_MOVIE_DETAILS_FAILURE 1`] = `
Object {
  "details": Object {
    "movieName": "test",
  },
}
`;

exports[`Reducer movie details should update store on GET_MOVIE_DETAILS_SUCCESS 1`] = `
Object {
  "details": Object {
    "movieName": "matrix",
  },
}
`;
